### ISSUE TITLE
LSP cannot find headers (C/C++)

### ISSUE THREAD

**Issue by laudrup** [emacs-lsp/lsp-mode#1311]
I'm having an issue getting LSP to find non-standard headers in a C++ project.

I've added the lsp packge in my emacs init as described in the documentation and that seems to just work, but LSP complains that it cannot find the header for any header not in the standard include path.

I've create a very simple example here:
[test_lsp.tar.gz](https://github.com/emacs-lsp/lsp-mode/files/4045409/test_lsp.tar.gz)

It is basically just a main function with a static library (and header) as a CMake target that generates a compilation database.

If I try to open main.cpp in emacs in lsp-mode, lsp complains that it cannot find "foo.h" even though the path to that is available in the compilation database that lsp seems to use.

Using another tool that also uses the compilation database (eg. clang-tidy) seems to work just fine, so I'm fairly certain that the compilation database is generated correctly (it looks fine as well).

Curiously, lsp/clang also complains about "#pragma once in main file" when opening the header file, but I do not know if these two things are related.

It is most likely me doing something wrong, but I'm not really sure how to debug my configuration.

Any ideas on how to figure out what I'm missing?

Using clang 9.0.1 on Debian GNU/Lunux 10 with the latest version of lsp-mode from elpa.

Thanks a lot and kind regards,

Kasper Laudrup

**Comment by laudrup** [comment]
I figured out what was wrong and of course it was related to my configuration/setup.

I had missed that I had enabled flycheck globally in my emacs init file (global-flycheck-mode).

Removing that solves the issue.

I have no idea why I had added that in the first place, but not enabling things globally is probably a good idea anyway :-)

Leaving this comment here in hope that it might help someone else in the future.

Thanks for the great work with lsp-mode!
